packages/core/src/browser/preferences/preference-contribution.ts

```diff
@@ -96,6 +85,26 @@ export class PreferenceSchemaProvider extends PreferenceProvider {
         return this.preferences;
     }
 
+    setSchema(schema: PreferenceSchema, updateValidation: boolean = true): void {
```

Comment:
How abou, instead of adding internal options to public APIs extract 2 methods:
```ts
setSchema(schema: PreferenceSchema): void {
    this.doSetSchema(schema);
    this.updateValidate();
}
protected doSetSchema() ... // updates combined schema
protected updateValidate() ... // updates validate function
```

and call `doSetSchema` and `updateValidate` from the constructor instead of `setSchema`?

Reply:
yes, this will be better, I'm updating the PR